lib/rototiller/utilities/command_flag.rb

```diff
@@ -32,4 +34,21 @@ def describe_flag_state
     flag_with_value = "The CLI flag #{@flag} will be used with value #{@value}."
     green_text(@value.nil? ? only_flag : flag_with_value)
   end
+
+  def validate_attribute_hash(h)
+    # validate the contents of the hash
+    name_error = 'A name must be supplied to add_flag'
+    value_error = 'A value must be supplied to add flag'
+    error = nil
+
+    if !h[:name] && !h[:value]
+      error = name_error << "\n" << value_error
+    elsif h[:name] && !h[:value]
+      error = value_error
+    elsif !h[:name] && h[:value]
+      error = name_error
+    end
+
+    raise(ArgumentError, error) if error
+  end
```

Comment:
hmmm.  we may want to consider the creation of a validator class for hash/block receiving methods.
we already kinda have a schema, somewhat defined in add_env() add_flag() in the attributes argument.
this code probably should have been replicated into env_var